### PR TITLE
Support disconnect

### PR DESCRIPTION
**Pull request by haoqi123**

### CODE REVIEW

File: iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/Mqtt.java [hunk 1]
```diff
@@ -32,6 +33,7 @@ private Mqtt(@Nonnull PulsarAgent pulsarAgent, @Nonnull IotPulsarMqttOptions opt
         // Automatic register
         processorController.register(MqttMessageType.CONNECT, new ConnectProcessor(this));
         processorController.register(MqttMessageType.PUBLISH, new PublishProcessor(this));
+        processorController.register(MqttMessageType.DISCONNECT, new DisConnectProcessor());
```
Comment (mattisonchao) [inline iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/Mqtt.java:36]: Typo: `Disconnect`?

File: iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/processor/DisConnectProcessor.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import io.iot.pulsar.mqtt.messages.custom.VoidMessage;
+import io.netty.handler.codec.mqtt.MqttMessage;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+import lombok.extern.slf4j.Slf4j;
+
+@ThreadSafe
+@Slf4j
+public class DisConnectProcessor  implements MqttProcessor {
```
Comment (mattisonchao) [inline iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/processor/DisConnectProcessor.java:13]:
```suggestion
public class DisConnectProcessor implements MqttProcessor {
```

File: iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/processor/DisConnectProcessor.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    @Nonnull
+    @Override
+    public CompletableFuture<MqttMessage> process(@Nonnull MqttEndpoint endpoint, @Nonnull MqttMessage message) {
+        endpoint.close();
```
Comment (mattisonchao) [inline iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/processor/DisConnectProcessor.java:17]: It's an async method. We should ensure the `voidMessage` return after the future is complete.

**Review by mattisonchao — CHANGES_REQUESTED** [review]

File: iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/Mqtt.java (hunk 1 above)
Comment (haoqi123) [reply iot-pulsar-mqtt/src/main/java/io/iot/pulsar/mqtt/Mqtt.java]: o~~~k

**Review by mattisonchao — APPROVED** [review]

**Pull request merged**